beep/featurize.py

```diff
@@ -595,17 +599,184 @@ def features_from_processed_cycler_run(cls, processed_cycler_run, params_dict=No
+    Object corresponding to summary statistics from
+    regular cycles
 
-   name (str): predictor object name.
-   X (pandas.DataFrame): features in DataFrame format.
-   metadata (dict): information about the conditions, data
-       and code used to produce features
-   """
+    name (str): predictor object name.
+    X (pandas.DataFrame): features in DataFrame format.
+    metadata (dict): information about the conditions, data
+        and code used to produce features
+    """
+
+    # Class name for the feature object
+    class_feature_name = "CycleSummaryStats"
+
+    def __init__(self, name, X, metadata):
+        """
+        Args:
+            name (str): predictor object name
+            X (pandas.DataFrame): features in DataFrame format.
+            metadata (dict): information about the data and code used to produce features
+        """
+        super().__init__(name, X, metadata)
+        self.name = name
+        self.X = X
+        self.metadata = metadata
+
+    @classmethod
+    def validate_data(cls, processed_cycler_run, params_dict=None):
+        """
+        This function determines if the input data has the necessary attributes for
+        creation of this feature class. It should test for all of the possible reasons
+        that feature generation would fail for this particular input data.
+
+        Args:
+            processed_cycler_run (beep.structure.ProcessedCyclerRun): data from cycler run
+            params_dict (dict): dictionary of parameters governing how the ProcessedCyclerRun object
+                gets featurized. These could be filters for column or row operations
+        Returns:
+            bool: True/False indication of ability to proceed with feature generation
+        """
+        if params_dict is None:
+            params_dict = FEATURE_HYPERPARAMS[cls.class_feature_name]
+
+        # TODO: not sure this is necessary
+        # Check for data in each of the selected cycles
+        index_1, index_2 = params_dict['cycle_comp_num']
+        cycle_1 = processed_cycler_run.cycles_interpolated[
+            processed_cycler_run.cycles_interpolated.cycle_index == index_1]
+        cycle_2 = processed_cycler_run.cycles_interpolated[
+            processed_cycler_run.cycles_interpolated.cycle_index == index_2]
+        if len(cycle_1) == 0 or len(cycle_2) == 0:
+            return False
+
+        # TODO: check whether this is good
+        # Check for relevant data
+        required_columns = ['charge_capacity',
+                            'discharge_capacity'
+                            # 'charge_energy',
+                            # 'discharge_energy'
+                            ]
+        pcycler_run_columns = processed_cycler_run.cycles_interpolated.columns
+        if not all([column in pcycler_run_columns for column in required_columns]):
+            return False
+
+        return True
+
+    STANDARD_OPERATION_NAMES = ["var", "min", "mean", "skew", "kurtosis", "abs", "square"]
+
```

Comment:
Suggest renaming this to `SUMMARY_STATISTICS` or something like it. Operation is a little over-loaded with physical battery operation :-)

Reply:
Agreed, used SUMMARY_STATISTIC_NAMES